src/accelgyrotemp_ros_publisher.cpp

```diff
@@ -60,13 +62,42 @@ void AccelGyroTempRosPublisher::run()
         switch(operation_mode) {
         case DEVICE_CONTINUOUS_SAMPLING_MODE:
             success = false;
-            success = m_dataProvider->getData(m_message);
 
-            RCLCPP_INFO(rclcpp::get_logger("rclcpp_accelgyrotemp"), "Publishing acceleration data: '%f' '%f' '%f'",
-                        m_message.accel.x, m_message.accel.y, m_message.accel.z);
+            bufferedDataSelection = m_node->get_parameter("buffered_data_selection").get_parameter_value().get<int32_t>();
 
-            if(success)
-                m_publisher->publish(m_message);
+            switch(bufferedDataSelection) {
+            case ACCEL_GYRO_BUFFERED_DATA:
+                if(bufferedDataEnabled)
```

Comment:
so bufferedDataEnabled means that the buffered data is enabled, which means you do not need to call enableBufferedDataOutput, please change the logic here.
with if (!bufferedDataEnabled)